include/aspect/material_model/visco_plastic.h

```diff
@@ -221,16 +227,86 @@ namespace aspect
          * compositional field.
          */
         std::pair<double, double>
-        calculate_weakening ( const double strain_ii,
-                              const unsigned int j ) const;
+        calculate_plastic_weakening ( const double strain_ii,
+                                      const unsigned int j ) const;
+
+        /**
+         * A function that computes the strain weakened values
+         * of the diffusion and dislocation prefactors for a given
+         * compositional field.
+         */
+        double
+        calculate_viscous_weakening ( const double strain_ii,
+                                      const unsigned int j ) const;
 
+        /**
+         * Whether to use the accumulated strain to weaken
+         * plastic parameters cohesion and friction angle
+         * and viscous parameters diffusion and dislocation prefactor.
```

Comment:
It might be worth making this wording a bit more specific. How about:

> Whether to use the accumulated total strain or plastic strain to weaken the cohesion and friction angle, and the viscous strain to weaken the refactors for diffusion and dislocation creep.